collect_app/src/test/java/org/odk/collect/android/utilities/TextUtilsTest.java

```diff
@@ -0,0 +1,33 @@
+package org.odk.collect.android.utilities;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricTestRunner;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+
+
+@RunWith(RobolectricTestRunner.class)
+public class TextUtilsTest {
+
+    /**
+     * Should return null if provided with null and not throw a NPE.
+     */
+    @Test
+    public void textToHtml_BounceNullInput() {
+        String input = null;
+        CharSequence observed = TextUtils.textToHtml(input);
+        assertNull(observed);
+    }
+
+    /**
+     * Should do something not-null-ish if provided with non-null string.
+     */
+    @Test
+    public void textToHtml_CallsHTMLOnRealInput() {
+        String input = "This *bold* markdown is \n- nice,\n- efficient.\n";
+        CharSequence result = TextUtils.textToHtml(input);
+        assertNotNull(result);
```

Comment:
@lindsay-stevens I appreciate the sentiment, but I can't claim special expertise over @srsudar. However, after looking closely at the code he commented on, I agree with his sentiment (especially with establishing the testing foundation in the code base), but I am not sure I like the alternative offered any better. 

My major complaint is that a good test usually _makes the issue involved clear_. To gauge the relevance of the test I had to reference #196. Unfortunately, the issue isn't clear on the root issue either. So, we (as will others in the future) have to lean on the test code. A good test method name consistent with a concise implementation should be enough. Sam's re-implementation brings the method contents back into alignment with the method name, but the method name does not help to understand the value of the test (other than a html method is called).

I'm still foggy on the purpose of the test, but is there a reason we're diving into the guts of the implementation? For example, could we create a focused test for the defect like so?
```java
@Test
public void singleMarkdownReplacement() 
    String input = "This *bold* markdown is \n- nice,\n- efficient.\n";
    assertEquals("unexpected markdown-to-html translation", "This <b>bold</b> markdown is \n- nice,\n- efficient.\n", TextUtils.textToHtml(input));
}
```

Obviously, this test *does not* help to understand #196, but it is a clear test and it is stable with a change in implementation since we treat the method as a black box. If you have a specific test case and a good description that uniquely corners #196 like this, it would be a significant improvement. If I get a chance, I will try to come up with one myself. 

Let me know if my feedback isn't clear.